light-cli/src/commands/history/index.ts

```diff
@@ -0,0 +1,69 @@
+import { Command, Flags } from "@oclif/core";
+import { IndexedTransaction } from "@lightprotocol/zk.js";
+import { CustomLoader, getUser } from "../../utils/utils";
+
+class TransactionHistoryCommand extends Command {
+  static description = "Retrieve transaction history for the user";
+
+  static flags = {
+    latest: Flags.boolean({
+      char: "l",
+      description: "Retrieve the latest transaction history",
+      default: true,
+    }),
+  };
+
+  protected finally(_: Error | undefined): Promise<any> {
+    process.exit();
+  }
+
+  static examples: Command.Example[] = [
+    "$ light history",
+    "$ light history --latest=false",
+  ];
+
+  async run() {
+    const { flags } = await this.parse(TransactionHistoryCommand);
+
+    const { latest } = flags;
+
+    const loader = new CustomLoader("Retrieving user transaction history...");
+
+    loader.start();
+
+    const user = await getUser();
+
+    try {
+      const transactions: IndexedTransaction[] =
+        await user.getTransactionHistory(latest);
+
+      // Log the transaction history
+      transactions.forEach((transaction) => {
+        this.log("\n--- Transaction ---");
+        this.log("Block Time:", transaction.blockTime);
+        this.log("Signer:", transaction.signer.toString());
+        this.log("Signature:", transaction.signature);
+        this.log("From:", transaction.from.toString());
+        this.log("To:", transaction.to.toString());
```

Comment:
how will this work for shielded transfer, you might not know the to only the from or vice versa

Reply:
also I think you can reuse a print utxo function